### PR TITLE
Add autoplot.cor_df method

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR attempts to add a `autoplot()` method to the `cor_df` objects that come of out `correlate()`. This function came out of a little chat I had with @juliasilge. By itself it has decent defaults with minimal code needed. Further suggestions are welcomed 

``` r
library(corrr)

mtcars |>
  correlate() |>
  autoplot(barheight = 10) 
```

It also defaults colors are also fairly colorblind friendly 

``` r
mtcars |>
  correlate() |>
  autoplot(barheight = 10) |>
  colorblindr::cvd_grid()

```

### CODE REVIEW

**Comment by thisisdaryn** [comment]
Hi @EmilHvitfeldt, 

Does/will this make `rplot()` obsolete?

**Comment by juliasilge** [comment]
My opinion is that we can keep the `rplot()` function as is, and add this one in as an additional plotting option. Some people might prefer the older option, but I agree with Emil that a more updated default-type plot would be nice. I don't think the maintenance burden will be too high. What do you think about that @thisisdaryn?

**Comment by thisisdaryn** [comment]
I don't have any problem with the new function. And `autoplot()` will be familiar to users of other packages. And I like the plots more than the current ones from `rplot()`.

I agree that the maintenance of both will not be too high.

It seems like `rplot()` does one thing and `autoplot()` will now be doing that one thing better. The only question I have is whether it would be useful to have an option for the method argument to keep the variables in the order they are in the data or to put them in alphabetical order. (I do think/agree that "PCA" is a sensible default.)

**Comment by EmilHvitfeldt** [comment]
You can turn off reordering by setting `method = "identity"`, and one could get alphabetic ordering by manually ordering being using `method = "identity"`.

``` r
library(corrr)
library(dplyr)

mtcars %>%
  correlate() %>%
  autoplot(method = "Identity")

```



``` r
mtcars %>%
  select(sort(colnames(.))) %>%
  correlate() %>%
  autoplot(method = "Identity")
```

**Comment by juliasilge** [comment]
Want to do a [code review](https://github.com/features/code-review) for this @thisisdaryn, checking out how it works, looking through the new docs, seeing if you have any thoughts, etc? I like [this resource for code review](https://google.github.io/eng-practices/review/reviewer/) quite a lot.

**Comment by thisisdaryn** [comment]
> Want to do a [code review](https://github.com/features/code-review) for this @thisisdaryn, checking out how it works, looking through the new docs, seeing if you have any thoughts, etc? I like [this resource for code review](https://google.github.io/eng-practices/review/reviewer/) quite a lot.

Sounds good. I'll work on it.

**Review by thisisdaryn — COMMENTED** [review]
I have two notes: 

1. I think a more natural default would be an upper triangular plot with the x axis text along the top as in the image. This would allow looking up correlations in a similar orientation to most printed tables.

2. There is a minor misspelling of the word "information" in correlate.R (line 32)

I am otherwise fine with the implementation and the end result.

**Comment by juliasilge** [comment]
@thisisdaryn I'm not totally sure about your first note. When I do an [image search for "correlation table"](https://www.google.com/search?q=correlation+table) it seems like most are actually a bottom triangular plot. Do you think that's not representative? Or have you had different experience?

**Comment by thisisdaryn** [comment]
Hi @juliasilge, I was not making a claim about the relative prevalence. I had done a [similar image search for "correlation matrix"](https://www.google.com/search?q=correlation+matrix&tbm=isch) and seen that lower triangular seemed to be more common than upper triangular but also that the full matrix and - in some instances - the lower right triangle were used in some examples. 

I found it easier/more natural to look up the tiles for a given variable pair in the upper triangular orientation (i.e. by shaving the other half, placing the x-axis text on top rotated by 315 degrees) while doing the review. Admittedly this is not very scientific or rigorous.   

That said, if we are going to go with lower triangular as the default, I would then suggest adding an argument to `autoplot()` to get an upper triangular matrix optionally (and perhaps also the full table/matrix).

**Comment by juliasilge** [comment]
@EmilHvitfeldt What do you think about making an argument for something like `triangular = c("upper", "lower", "full")`?

**Comment by EmilHvitfeldt** [comment]
That would be doable! I'll get right on it

**Comment by EmilHvitfeldt** [comment]
Should be done now 🙌 

``` r
library(corrr)
x <- correlate(mtcars)

autoplot(x)
```



``` r
autoplot(x, triangular = "lower")
```



``` r
autoplot(x, triangular = "full")
```



<sup>Created on 2022-04-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

File: R/cor_df.R [hunk 1]
```diff
@@ -353,19 +356,33 @@ network_plot.cor_df <- function(rdf,
...
-                            low = "#B2182B", mid = "#F1F1F1",
+autoplot.cor_df <- function(object, ...,
+                            method = "PCA",
+                            triangular = c("upper", "lower", "full"),
+                            barheight = 20,
+                            low = "#B2182B",
+                            mid = "#F1F1F1",
                             high = "#2166AC") {
 
+  triangular <- match.arg(triangular)
```
Comment (juliasilge) [inline R/cor_df.R:372]:
```suggestion
  triangular <- rlang::arg_match(triangular)
```

File: R/cor_df.R [hunk 2]
```diff
@@ -336,6 +336,9 @@ network_plot.cor_df <- function(rdf,
 #' @param object A `cor_df` object.
 #' @param ... this argument is ignored.
 #' @inheritParams rearrange
+#' @param triangular a single character, which part of the correlation matrix
+#' should be shown? Must be one of `"upper"`, `"lower"`, or `"full"`. Defaults
+#' to `"upper"`.
```
Comment (juliasilge) [inline R/cor_df.R:341]:
```suggestion
#' @param triangular Which part of the correlation matrix should be shown?
#' Must be one of `"upper"`, `"lower"`, or `"full"`, and defaults to `"upper"`.
```

**Review by juliasilge — APPROVED** [review]

**Pull request merged**